internal/controller/controllers/kube_compat.go

```diff
@@ -18,3 +33,168 @@ func ServerIsOpenShift(ctx context.Context, c client.Client) (bool, error) {
 	}
 	return false, client.IgnoreNotFound(err)
 }
+
+func serverSupportsCertManager(ctx context.Context, c client.Client) (bool, error) {
+	certCRD := apiextensionsv1.CustomResourceDefinition{}
+	err := c.Get(ctx, types.NamespacedName{Name: certificateCRDName}, &certCRD)
+	if err == nil {
+		return true, nil
+	}
+	return false, client.IgnoreNotFound(err)
```

Comment:
nit: this could probably be inlined by reversing the condition

Reply:
Not sure what you mean but I guess something like this would work:
```suggestion
	err := c.Get(ctx, types.NamespacedName{Name: certificateCRDName}, &certCRD)
	return err == nil, client.IgnoreNotFound(err)
```

Is that what you're suggesting?

Reply:
I meant
```golang
if err := c.Get(ctx, types.NamespacedName{Name: certificateCRDName}, &certCRD); err != nil {
    return false, client.IgnoreNotFound(err)
}
return return true, nil
```